gis_exercise_2.py

```diff
@@ -178,7 +178,19 @@
 # hyperlink @sort: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
```

Comment:
brilliant! But also quite disheartening, I would have never achieved this.

Reply:
You would've eventually. I've been working with data for over 10 years, so I do it quickly, but it takes time to wrap your head around all the concepts. 

There are a lot of concepts to build upon here: data manipulation + dataframes + geo + functional manipulation of data (map, filter, and reduce functions) + first order functions + .... etc

Reply:
Gonna take advantage if I have you on the line right now:

How would I access the userid of specific values in your code? (e.g. of the max distance travelled of a user). I thought it would be super easy, but I am a bit dumbfounded by what I find in the internet.

Reply:
nvm, I think I figured it out. I guess using iteritems() is very inefficient, but it is what I have found (Baby steps :)) )

Reply:
haha, I depends on the details of what you're doing it. `iteritems()` is actually very efficient in the sense that it iterates over the items one by one, and you don't need to get the whole list, so you could do: `first = next(frame.iteritems())` to get the first one. If you iterate over the whole dataframe though, then yeah, *very* inneficient